### ISSUE TITLE
Build leaves 'localpkg' files in src tree

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#8130]
A clean build leaves the following files in the `src` tree (according to `git status`):
```
        src/.nuget/Microsoft.NETCore.ILAsm/localpkg/
        src/.nuget/Microsoft.NETCore.ILAsm/win/localpkg/
        src/.nuget/Microsoft.NETCore.ILDAsm/localpkg/
        src/.nuget/Microsoft.NETCore.ILDAsm/win/localpkg/
        src/.nuget/Microsoft.NETCore.Jit/localpkg/
        src/.nuget/Microsoft.NETCore.Jit/win/localpkg/
        src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/localpkg/
        src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/win/localpkg/
        src/.nuget/Microsoft.NETCore.TestHost/localpkg/
        src/.nuget/Microsoft.NETCore.TestHost/win/localpkg/
        src/.nuget/Microsoft.TargetingPack.Private.CoreCLR/localpkg/
```
Should these either be hidden by adding to `.gitignore`, or should they be created in the `bin` directory for the build flavor instead?

Bottom-line: they shouldn't appear in `git status` output.

**Comment by adityamandaleeka** [comment]
@wtgodbe Is this related to the package build stuff you worked on?

**Comment by gkhanna79** [comment]
@wtgodbe PTAL.

**Comment by wtgodbe** [comment]
I don't think this is related to my change, that only affected how we restore packages, this looks more related to how we build them. I can take a closer look when I'm back from vacation in a week.

**Comment by jamesqo** [comment]
This is also happening for me...

**Comment by JosephTremoulet** [comment]
I'm not sure why adding `localpkg` was fine in #8791/#8882 when it wasn't fine in #8343... are we happy with this fix, should this issue be closed, or do we want to come back with a different fix?

/cc @janvorli @jkotas

**Comment by janvorli** [comment]
@JosephTremoulet my comment in #8343 described the reason. I was thinking that we should fix the problem of the files being generated into the source folder rather than just hiding it by adding the git ignore. The comment here indicated the possible fix that already happened in the tools repo:
https://github.com/dotnet/coreclr/pull/8791/files#r94514279

However, I have found in the meanwhile that it is not that straightforward to just update the tools to the latest version, so I am ok with the git ignore as a temporary fix until the tools are fixed.

**Comment by gkhanna79** [comment]
@wtgodbe PTAL at this when you can.

**Comment by janvorli** [comment]
I have actually given it a quick look and experimented a bit. And the reason for the problem is simple. In the ./Tools/Packaging.targets, $(BaseOutputPath) is used as the target folder for the localpkg folder. But, the $(BaseOutputPath) is not defined! I have verified that by putting hardcoded path there instead, which resulted in the localpkg being created in that folder and I have also sticked the $(BaseOutputPath) in the middle of my path and it was still created in my folder, proving that the $(BaseOutputPath) is not defined.
In other words, both of the following two versions of the same line generate the files into /home/janvorli/git/coreclr/bin/obj/Linux.x64.Debug/localpkg/.

```xml
<PackageInstallPath Condition="'$(PackageInstallPath)' == ''">/home/janvorli/git/coreclr/bin/obj/Linux.x64.Debug/localpkg/</PackageInstallPath>

<PackageInstallPath Condition="'$(PackageInstallPath)' == ''">/home/janvorli/git/coreclr/bin/obj/Linux.x64.Debug/$(BaseOutputPath)localpkg/</PackageInstallPath>
```

**Comment by wtgodbe** [comment]
@janvorli did you put up a fix for this yesterday? I can put together a PR if not.

**Comment by gkhanna79** [comment]
I just did a build on Mac and I dont see this issue. Is this Linux specific?

**Comment by jkotas** [comment]
I see it on Windows all the time. nuget packaging build step creates these.

**Comment by janvorli** [comment]
The issue is not visible because someone has already added the lockapkg to the .gitignore. You just need to remove it and you'll see it right away.

**Comment by wtgodbe** [comment]
https://github.com/dotnet/coreclr/pull/8934 should resolve

**Comment by JosephTremoulet** [comment]
Yes, fixed in #8934, thanks.
